### ISSUE TITLE
pFocusTrap: "aria-hidden-focus" accessibility rule violation

### ISSUE THREAD

**Issue by andrey-post** [primefaces/primeng#16042]
### Describe the bug

https://github.com/primefaces/primeng/issues/15482
New implementation of `pFocusTrap` has `aria-hidden-focus` accessibility rule violation:
![image](https://github.com/user-attachments/assets/424ac91e-3971-4a18-89c3-bc2eac66c96f)

### Environment

![image](https://github.com/user-attachments/assets/9f354c0e-cf48-4394-9520-73f521474a17)

### Angular version

17.3.11

### PrimeNG version

17.18.4

### Build / Runtime

TypeScript

### Language

TypeScript

### Node version (for AoT issues node --version)

18.20.2

### Steps to reproduce the behavior

1. Go to https://primeng.org/focustrap
2. Install [Axe DevTools](https://chromewebstore.google.com/detail/axe-devtools-web-accessib/lhdoppojpmngadmnindnejefpokejbdd) accessibility scanner extension
3. Open dev tools and run Full Page check
4. Observe `aria-hidden-focus` rule violation by invisible span elements added by `pFocusTrap` directive

https://dequeuniversity.com/rules/axe/4.9/aria-hidden-focus?application=AxeChrome

### Expected behavior

`pFocusTrap` should work without accessibility rules violations.
